safesocket/pipe_windows.go

```diff
@@ -26,20 +26,46 @@ func setFlags(network, address string, c syscall.RawConn) error {
 	})
 }
 
-// TODO(apenwarr): use named pipes instead of sockets?
-//
-//	I tried to use winio.ListenPipe() here, but that code is a disaster,
-//	built on top of an API that's a disaster. So for now we'll hack it by
-//	just always using a TCP session on a fixed port on localhost. As a
-//	result, on Windows we ignore the vendor and name strings.
-//	NOTE(bradfitz): Jason did a new pipe package: https://go-review.googlesource.com/c/sys/+/299009
 func listen(path string, port uint16) (_ net.Listener, gotPort uint16, _ error) {
-	lc := net.ListenConfig{
-		Control: setFlags,
+	// Construct the default named pipe security descriptor.
+	var acl *windows.ACL
+	if err := windows.RtlDefaultNpAcl(&acl); err != nil {
+		return nil, 0, err
+	}
+	defer windows.LocalFree(windows.Handle(unsafe.Pointer(acl)))
+	sd, err := windows.NewSecurityDescriptor()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	allUsers, err := windows.CreateWellKnownSid(windows.WinBuiltinUsersSid)
```

Comment:
We might consider using `Network Configuration Operators` which is a standard group that may be a better fit.

Reply:
That group does exist (it's an alias to other built-in stuff), but I'm fine with this code as-is for the time being.

Perhaps my only suggestion is that, since this ACL involves a well-known group, we don't need to go through the whole `ACLFromEntries` rigamarole every time -- we could just put the serialized SDDL string into the code as a constant and pass that straight into `PipeConfig`.

Reply:
Done, PTAL